presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaSecurityConfig.java

```diff
@@ -0,0 +1,142 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.configuration.LegacyConfig;
+import io.airlift.configuration.validation.FileExists;
+import org.apache.kafka.common.config.SslConfigs;
+
+import java.util.Locale;
+import java.util.Properties;
+
+/**
+ *  {@KafkaSecurityConfig} manages Kafka configuration related to connection security. I.e. if the security protocol
+ *  'SSL' is being used. If so, additional properties might be supplied, i.e. location and password of the truststore
+ *  and keystore to be used.
+ */
+public class KafkaSecurityConfig
+{
+    private Properties props;
+
+    private String sslKeystoreFile;
+    private String sslKeystorePassword;
+    private String sslKeyPassword;
+    private String sslTruststoreFile;
+    private String sslTruststorePassword;
+    private KafkaEndpointIdentificationAlgorithm sslEndpointIdentificationAlgorithm;
+
+    @Config("kafka.keystore-file")
+    @LegacyConfig("kafka.ssl.keystore.location")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_LOCATION_DOC)
+    public void setSslKeystoreFile(String sslKeystoreFile)
+    {
+        this.sslKeystoreFile = sslKeystoreFile;
+    }
+
+    @FileExists
+    public String getSslKeystoreFile()
+    {
+        return sslKeystoreFile;
+    }
+
+    @Config("kafka.keystore-password")
+    @LegacyConfig("kafka.ssl.keystore.password")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslKeystorePassword(String sslKeystorePassword)
+    {
+        this.sslKeystorePassword = sslKeystorePassword;
+    }
+
+    public String getSslKeystorePassword()
+    {
+        return sslKeystorePassword;
+    }
+
+    @Config("kafka.key-password")
+    @LegacyConfig("kafka.ssl.key.password")
+    @ConfigDescription(SslConfigs.SSL_KEY_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslKeyPassword(String sslKeyPassword)
+    {
+        this.sslKeyPassword = sslKeyPassword;
+    }
+
+    public String getSslKeyPassword()
+    {
+        return sslKeyPassword;
+    }
+
+    @Config("kafka.truststore-file")
+    @LegacyConfig("kafka.ssl.truststore.location")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_LOCATION_DOC)
+    public void setSslTruststoreFile(String sslTruststoreFile)
+    {
+        this.sslTruststoreFile = sslTruststoreFile;
+    }
+
+    @FileExists
```

Comment:
`@NotNull`?

Reply:
`@FileExists` was explicitly recommended by `@losipiuk`... If no truststore should be set, the String might remain `null` and nothing will be forwarded to `kafka-clients`

Reply:
Is this optional property when SSL is used?

Reply:
All of the SSL related properties are optional...

Reply:
Does it mean that user can require SSL protocol and still not specify them? Or user may use SSL or not, but when it is used all SSL related properties are required and when no SSL is requested then all SSL properties should not be used.

Reply:
Yes, I checked how the underlying `kafka-clients` library handles this (since I use that in other projects/applications):
- You can set `security.protocol=PLAINTEXT` and still specify a keystore + password -> They will show up in the configuration used by the library if you let it log that, but I would assume that it does not really use it at the end of the day...
- You can also set `security.protocol=SSL` and do not specify any additional stuff: Then the JVM default truststore will be used...

In my case I have one cluster for which I need to use `SSL` and a specific truststore+password in order to trust the certificate used in the broker; for another cluster I also need to provide a keystore with a client certificate+private key since that cluster enforces the use of client certificates...

Reply:
> You can set security.protocol=PLAINTEXT and still specify a keystore + password -> They will show up in the configuration used by the library if you let it log that, but I would assume that it does not really use it at the end of the day...

I guess this would be very misleading to the user. I would prefer to disallow such configuration. i.e. when `security.protocol=PLAINTEXT` we should not allow to pass any ssl specific properties.

> You can also set security.protocol=SSL and do not specify any additional stuff: Then the JVM default truststore will be used...

That makes sense.

Can you connect to broker with SSL without keystore?

Reply:
> I guess this would be very misleading to the user. I would prefer to disallow such configuration. i.e. when security.protocol=PLAINTEXT we should not allow to pass any ssl specific properties.

I had this as part of my code initially, had this implemented - I logged a warning that a given property wasn't used due to the fact that `PLAINTEXT` is being used... I think this was removed in succession of multiple review remarks...

Should I re-add that? If so, should I keep the security related stuff separate or merge it back into KafkaConfig (where the `broker.nodes` property resides to which those security related properties belong?

> Can you connect to broker with SSL without keystore?

Yes, sure - if the cluster does not require client certificates for verification, not keystore is needed (also the private key, keystore and truststore do not need to be password protected...)